### ISSUE TITLE
Investigate API error for the Demo site Federal Test Service

### ISSUE THREAD

**Issue by ccostino** [GSA/notifications-api#798]
When navigating to the Usage page for the the `Federal Test Service` in the Demo Notify.gov site, the page returns an error, which is ultimately being thrown by the API.

This is the stack trace I saw in the logs:

```
Traceback (most recent call last):
  File "/home/vcap/deps/0/python/lib/python3.9/site-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/vcap/deps/0/python/lib/python3.9/site-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/vcap/deps/0/python/lib/python3.9/site-packages/newrelic/hooks/framework_flask.py", line 82, in _nr_wrapper_handler_
    return wrapped(*args, **kwargs)
  File "/home/vcap/app/app/service/rest.py", line 570, in get_monthly_notification_stats
    statistics.add_monthly_notification_status_stats(data, stats)
  File "/home/vcap/app/app/service/statistics.py", line 116, in add_monthly_notification_status_stats
    data[month][row.notification_type][row.notification_status] += row.count
KeyError: '2023-01'
```

It looks like it's a data issue and something up with the service itself, but we're not handling it properly within the code.  Hence the exception being thrown, which is a less than ideal customer experience if it were to happen in production.

## Implementation Sketch and Acceptance Criteria

- [ ] Investigate the cause of the error and missing key being referenced
- [ ] Add proper data and exception handling to the appropriate methods
- [ ] Add or modify tests to cover the condition

## Security Considerations

- None with this change, it's not exposing any sensitive information or anything, but we'd prefer to not have unknown exceptions being thrown by the application.

**Comment by A-Shumway42** [comment]
I saw this bug while working through the `free_sms_allowance` bugs on the dashboard and ultimately the usage page.
This PR in the api has a change to the `get_months_for_financial_year(year)` method: https://github.com/GSA/notifications-api/pull/794 
It originally was based off the UK's financial year. Changed it to return months for the current calendar year. So, the `"month"` key in `data` will always be there, it will just have empty values. 
This is the PR for the front-end: https://github.com/GSA/notifications-admin/pull/1195
